### DISCUSSION TITLE
How to tell if isScrolling was triggered by followOutput or user scroll?

### DISCUSSION THREAD

**Discussion by gajus** [petyosi/react-virtuoso#500] [Q&A] [answered]
I want to implement behaviour that sets `followOutput` to `false` if user takes over scrolling. 

One way to do this is to capture `scrollerRef` and inside of the `isScrolling` check if scroll is at the bottom, e.g.

```ts
scrollerElement.scrollHeight - scrollerElement.scrollTop - scrollerElement.clientHeight
```

However, this will occasionally produce a false-positive.

**Comment by petyosi** [comment, marked as answer]
I feel like you're doing something either very exotic or your setup is something which I haven't seen. The behavior you describe is by default for `followOutput` (if I understand you right).  See [this demo](https://virtuoso.dev/stick-to-bottom/).

**Reply by gajus** [reply]
Interesting. In my case it just sticks to bottom.

`followOutput` is set to:

```ts
const followOutput = useCallback(() => {
  return followingOutput ? 'auto' : false;
}, [followingOutput]);
```

**Reply by petyosi** [reply]
Check [the docs on `followOutput`.](https://virtuoso.dev/virtuoso-api-reference/)  The callback receives an argument.

**Reply by gajus** [reply]
Awesome. My use case is addressed with:

```ts
const followOutput = useCallback((isAtBottom) => {
  if (followingOutput && !isAtBottom) {
    setFollowingOutput(false);

    return false;
  }

  return followingOutput ? 'auto' : false;
}, [
  followingOutput,
  setFollowingOutput,
]);
```

Thank you!
